django/utils/crypto.py

```diff
@@ -63,7 +63,7 @@ def get_random_string(length=12,
         # is better than absolute predictability.
         random.seed(
             hashlib.sha256(
-                ('%s%s%s' % (random.getstate(), time.time(), settings.SECRET_KEY)).encode()
+                ('%s%s%s' % (random.getstate(), time.monotonic(), settings.SECRET_KEY)).encode()
```

Comment:
I'm a bit worried that `monotonic()` makes this input a tiny bit more predictable than the non-consistent `time()`.

Reply:
Hey @MarkusH :wave: 

Can you elaborate on that more?

The result from `time.monotonic()` is system/machine specific (can depend on uptime for example) and will vary across machines. To me that seems to be generally much harder to predict than predicting a Unix timestamp.

Reply:
Hey @suligap :wave:, good to read you here :slightly_smiling_face: 

So, `monotonic()` guarantees a monotonic (how surprising) and constant increase in its value. If you were to call `monotonic()` twice, exactly 1 second apart, you'd get two values that are exactly `1` second apart (`call1 + 1 == call2`). `time()` does have this monotonic guarantee, as it also doesn't account for leap seconds, etc (like, all the fun things why one should use `monotonic()` to measure duration). With that in mind, we _do_ want to have "randomness" and unpredictability in the case at hand to generate the random seed. That's why I'm trying to argue that `monotonic()` might be more predictable than `time()`. But that's nothing more than a gut feeling, I don't have prove.